play-akka-http-ws-standalone/src/main/scala/play/api/libs/ws/akkahttp/StandaloneAkkaHttpWSRequest.scala

```diff
@@ -325,6 +326,27 @@ final class StandaloneAkkaHttpWSRequest private (
    * Execute this request
    */
   override def execute(): Future[Response] = {
+    def requestToResponse(request: HttpRequest): Future[HttpResponse] =
+      Http().singleRequest(request, connectionContext = ctx)
+
+    def handleRedirects(redirectsLeft: Int, request: HttpRequest)(response: HttpResponse): Future[HttpResponse] = {
+      if (redirectsLeft < 1) {
+        Future.failed(new IllegalStateException("Maximum redirect reached: " + MaxRedirects))
+      } else if (response.status.isRedirection() && config.followRedirects) {
+        val location = response.header[Location].get.uri
```

Comment:
Probably needs explicit handling of missing `Location` header to improve the error message in that case.